pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -0,0 +1,239 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+
+//go:generate mockgen -source $GOFILE -package=$GOPACKAGE -destination=generated_mock_$GOFILE
+
+package network
+
+import (
+	"fmt"
+
+	"github.com/vishvananda/netlink"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/log"
+	"kubevirt.io/kubevirt/pkg/precond"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+var bridgeFakeIP = "169.254.75.86/32"
+
+type BindMechanism interface {
+	discoverPodNetworkInterface() error
+	preparePodNetworkInterfaces() error
+	startDHCPServer()
+	decorateInterfaceConfig(ifconf *api.Interface)
+}
+
+type PodInterface struct{}
+
+func (l *PodInterface) Unplug() {}
+
+// Plug connect a Pod network device to the virtual machine
+func (l *PodInterface) Plug(iface *v1.Interface, network *v1.Network, domain *api.Domain) error {
+	precond.MustNotBeNil(domain)
+	initHandler()
+
+	driver, err := getBinding(iface)
+	if err != nil {
+		return err
+	}
+
+	// There should alway be a pre-configured interface for the default pod interface.
```

Comment:
It still might be worth double checking that there is one and returning an error. Otherwise this will panic.